shared/clarifai_shared/noun_phrase_extraction/extractor.py

```diff
@@ -0,0 +1,476 @@
+"""
+Noun phrase extractor for Claims and Summary nodes.
+
+This module implements the main NounPhraseExtractor class that:
+1. Fetches (:Claim) and (:Summary) nodes from Neo4j
+2. Extracts noun phrases using spaCy
+3. Normalizes and embeds the phrases
+4. Stores them in the concept_candidates vector table
+"""
+
+import logging
+import re
+import time
+from typing import List, Dict, Any, Optional, Set
+from dataclasses import dataclass
+
+try:
+    import spacy
+    from spacy.tokens import Doc
+except ImportError:
+    spacy = None
+    Doc = None
```

Comment:
More Explicit ImportError Handling for spaCy
Observation: The code handles a missing spacy library with a top-level try...except ImportError.
Issue: If the import fails, the NounPhraseExtractor will fail later at runtime with a NameError because spacy is not defined.

This should be a hard requirement, not wrapped in a try/except block.